### ISSUE TITLE
Error after adding swagger support 

### ISSUE THREAD

**Issue by sinner** [Romakita/ts-express-decorators#227]
## Description
I'm having these errors (**TypeError: Cannot set property 'id' of undefined** and **TypeError: Cannot read property 'merge' of undefined**) after adding the swagger configuration options:

## Evidence
The following is the console output:

```sh
[2018-01-31T19:35:06.105] [INFO ] [TSED] - Scan files : /var/www/javascript/MyProject/src/converters/**/*.ts
TypeError: Cannot read property 'merge' of undefined
    at Schema.decoratorSchemaFactory_1.decoratorSchemaFactory (/var/www/javascript/MyProject/node_modules/ts-express-decorators/src/jsonschema/decorators/schema.ts:43:16)
    at parameters (/var/www/javascript/MyProject/node_modules/ts-express-decorators/src/jsonschema/utils/decoratorSchemaFactory.ts:25:24)
    at Title.args (/var/www/javascript/MyProject/node_modules/ts-express-decorators/src/swagger/decorators/title.ts:65:46)
    at DecorateProperty (/var/www/javascript/MyProject/node_modules/ts-express-decorators/node_modules/reflect-metadata/Reflect.ts:1214:35)
    at Object.decorate (/var/www/javascript/MyProject/node_modules/ts-express-decorators/node_modules/reflect-metadata/Reflect.ts:737:24)
    at __decorate (/var/www/javascript/MyProject/src/controllers/current/CalendarCtrl.ts:4:92)
    at Object.<anonymous> (/var/www/javascript/MyProject/src/controllers/current/CalendarCtrl.ts:53:1)
    at Module._compile (module.js:643:30)
    at Module.m._compile (/var/www/javascript/MyProject/node_modules/ts-node/src/index.ts:392:23)
    at Module._extensions..js (module.js:654:10)
(node:21958) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): TypeError: Cannot read property 'merge' of undefined
(node:21958) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[nodemon] clean exit - waiting for changes before restart

[2018-01-31T18:37:30.791] [INFO ] [TSED] - Swagger Json is available on http://0.0.0.0:3080/api-docs/swagger.json
TypeError: Cannot set property 'id' of undefined
    at keys.forEach (/var/www/javascript/MyProject/node_modules/ts-express-decorators/src/swagger/class/OpenApiParamsBuilder.ts:117:37)
    at Array.forEach (<anonymous>)
    at OpenApiParamsBuilder.createSchema (/var/www/javascript/MyProject/node_modules/ts-express-decorators/src/swagger/class/OpenApiParamsBuilder.ts:116:14)
    at _parameters.injectedParams.map (/var/www/javascript/MyProject/node_modules/ts-express-decorators/src/swagger/class/OpenApiParamsBuilder.ts:47:73)
    at Array.map (<anonymous>)
    at OpenApiParamsBuilder.build (/var/www/javascript/MyProject/node_modules/ts-express-decorators/src/swagger/class/OpenApiParamsBuilder.ts:31:14)
    at OpenApiEndpointBuilder.build (/var/www/javascript/MyProject/node_modules/ts-express-decorators/src/swagger/class/OpenApiEndpointBuilder.ts:41:14)
    at ctrl.endpoints.forEach (/var/www/javascript/MyProject/node_modules/ts-express-decorators/src/swagger/services/SwaggerService.ts:186:83)
    at Array.forEach (<anonymous>)
    at SwaggerService.buildRoutes (/var/www/javascript/MyProject/node_modules/ts-express-decorators/src/swagger/services/SwaggerService.ts:182:24)
(node:16077) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): TypeError: Cannot set property 'id' of undefined
(node:16077) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[nodemon] clean exit - waiting for changes before restart
```

## Informations

Type | Version
 Bug |  3.9.1

****

A few sentences describing the overall goals of the issue.

## Todos

- [ ] Example (log, usecase, etc...)
- [x] Ts.ED version (bug/question)

**Comment by sinner** [comment]
Server Configuration:

```typescript

import "ts-express-decorators/swagger";
import {ServerLoader, ServerSettings, GlobalAcceptMimesMiddleware} from "ts-express-decorators";

@ServerSettings({
    rootDir,
    acceptMimes: ["application/json"],
    debug: false,
    httpPort,
    httpsPort,
    httpsOptions: {
        key: sslKeyFile,
        cert: sslCertFile,
        passphrase: sslPassPhrase
    },
    swagger: {
        path: "/api-docs"
    },
    mount: {
        "/api": `${rootDir}/controllers/current/**/*.ts`
    }
})
export class Server extends ServerLoader {
...
```

**Comment by sinner** [comment]
## Controller:

```typescript
import {Title, Description, Summary, Returns} from "ts-express-decorators/swagger";

@Controller("/calendars")
export class CalendarCtrl {

    @Get("/")
    @Title("Title of this route")
    @Summary("Summary of this route")
    @Description("Description of this route")
    @Returns(404, {description: "Not found"})
    async renderCalendars(request: Express.Request, response: Express.Response): 
    Promise<Array<Calendar>> {
        return [{id: '1', name: "test"}];
    }

    ...
```

**Comment by Romakita** [comment]
Hi @sinner ,
I've found the error. Normally Title decorator can't be used to describe a title method. This key doesn't exists in the Swagger spec.

However, I'll fix the Title decorator so that it doesn't produce an error in this case.

See you,
Romain

**Comment by sinner** [comment]
Hi Romain!

Thank you very much for your answer.

However, I've updated to the version 3.9.2 and I'm still having the error "TypeError: Cannot set property 'id' of undefined", even removing the @Title decorator from my controller.

**Comment by Romakita** [comment]
Hello @sinner,

Can you send me your log please. I can't reproduce your bug :/

See you
Romain

**Comment by sinner** [comment]
Hi @Romakita,

Of course, I'm still having the same configuration on the Server class and on the CalendarCtrl class.

The log is the following:

```sh
[2018-02-03T19:57:02.366] [INFO ] [TSED] - Swagger Json is available on http://0.0.0.0:3080/api-docs/swagger.json
TypeError: Cannot set property 'id' of undefined
    at keys.forEach (/Users/jsinner/Workspace/Javascript/Web/seed-typescript-express/node_modules/ts-express-decorators/src/swagger/class/OpenApiParamsBuilder.ts:117:37)
    at Array.forEach (<anonymous>)
    at OpenApiParamsBuilder.createSchema (/Users/jsinner/Workspace/Javascript/Web/seed-typescript-express/node_modules/ts-express-decorators/src/swagger/class/OpenApiParamsBuilder.ts:116:14)
    at _parameters.injectedParams.map (/Users/jsinner/Workspace/Javascript/Web/seed-typescript-express/node_modules/ts-express-decorators/src/swagger/class/OpenApiParamsBuilder.ts:47:73)
    at Array.map (<anonymous>)
    at OpenApiParamsBuilder.build (/Users/jsinner/Workspace/Javascript/Web/seed-typescript-express/node_modules/ts-express-decorators/src/swagger/class/OpenApiParamsBuilder.ts:31:14)
    at OpenApiEndpointBuilder.build (/Users/jsinner/Workspace/Javascript/Web/seed-typescript-express/node_modules/ts-express-decorators/src/swagger/class/OpenApiEndpointBuilder.ts:41:14)
    at ctrl.endpoints.forEach (/Users/jsinner/Workspace/Javascript/Web/seed-typescript-express/node_modules/ts-express-decorators/src/swagger/services/SwaggerService.ts:186:83)
    at Array.forEach (<anonymous>)
    at SwaggerService.buildRoutes (/Users/jsinner/Workspace/Javascript/Web/seed-typescript-express/node_modules/ts-express-decorators/src/swagger/services/SwaggerService.ts:182:24)
(node:15430) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): TypeError: Cannot set property 'id' of undefined
(node:15430) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[nodemon] clean exit - waiting for changes before restart
```

If it's useful:

+ nodejs@8.9.4
+ nodemon@1.14.12
+ @types/express@4.11.1
+ typescript@2.7.1
+ ts-express-decorators@3.10.0

**Comment by sinner** [comment]
Wonderful!
